tools/cmake/cores/Cortex-M4F.cmake

```diff
@@ -2,44 +2,26 @@
 # SPDX-License-Identifier: Apache-2.0
 
 # Sets cpu core options
-function(mbed_set_cpu_core_options target mbed_toolchain)
-    if(${mbed_toolchain} STREQUAL "GCC_ARM")
-        list(APPEND common_toolchain_options
-            "-mthumb"
-            "-mcpu=cortex-m4"
-            "-mfpu=fpv4-sp-d16"
-            "-mfloat-abi=softfp"
-        )
-
-        target_compile_options(${target}
-            INTERFACE
-                ${common_toolchain_options}
-        )
-
-        target_link_options(${target}
-            INTERFACE
-                ${common_toolchain_options}
-        )
-    elseif(${mbed_toolchain} STREQUAL "ARM")
-        list(APPEND compile_options
-            "-mcpu=cortex-m4"
-            "-mfpu=fpv4-sp-d16"
-            "-mfloat-abi=hard"
-        )
-
-        target_compile_options(${target}
-            INTERFACE
-                $<$<COMPILE_LANGUAGE:C>:${compile_options}>
-                $<$<COMPILE_LANGUAGE:CXX>:${compile_options}>
-                $<$<COMPILE_LANGUAGE:ASM>:-mcpu=Cortex-M4>
-        )
-
-        target_link_options(${target}
-            INTERFACE
-                "--cpu=Cortex-M4"
-        )
-    endif()
-endfunction()
+if(${MBED_TOOLCHAIN} STREQUAL "GCC_ARM")
+    list(APPEND common_options
+        "-mthumb"
+        "-mcpu=cortex-m4"
+        "-mfpu=fpv4-sp-d16"
+        "-mfloat-abi=softfp"
+    )
+elseif(${MBED_TOOLCHAIN} STREQUAL "ARM")
+    list(APPEND c_cxx_compile_options
+        "-mcpu=cortex-m4"
+        "-mfpu=fpv4-sp-d16"
+        "-mfloat-abi=hard"
+    )
+    list(APPEND asm_compile_options
+        "-mcpu=Cortex-M4"
```

Comment:
this is probably causing ARMClang to fail as , the command for startup file for K64F (asm file) contains `-mcpu=Cortex-M4-mcpu=cortex-m4`